### PR TITLE
TestTimeout in RunSettings

### PR DESCRIPTION
**Pull request by singhsarab**
Making test timeout configurable via runsettings
https://github.com/Microsoft/testfx-docs/pull/40

### CODE REVIEW

File: test/UnitTests/MSTest.CoreAdapter.Unit.Tests/Execution/TypeCacheTests.cs [hunk 1]
```diff
@@ -815,6 +816,57 @@ public void GetTestMethodInfoShouldThrowWhenTimeoutIsIncorrect()
             Assert.AreEqual(expectedMessage, exception.Message);
         }
 
+        [TestMethodV1]
+        public void GetTestMethodInfoWhenTimeoutAttributeNotSetShouldReturnTestMethodInfoWithGlobalTimeout()
+        {
+            string runSettingxml =
+                @"<RunSettings>
+                    <MSTestV2>
+                        <TestTimeout>4000</TestTimeout>
```
Comment (jayaranigarg) [inline test/UnitTests/MSTest.CoreAdapter.Unit.Tests/Execution/TypeCacheTests.cs:825]: Should also add one negative test case where timeout is invalid(negative value or non-integer value)

File: src/Adapter/MSTest.CoreAdapter/Execution/TypeCache.cs [hunk 2]
```diff
@@ -645,6 +646,10 @@ private int GetTestTimeout(MethodInfo methodInfo, TestMethod testMethod)
 
                 return timeoutAttribute.Timeout;
             }
+            else if (globalTimeout > 0)
```
Comment (jayaranigarg) [inline src/Adapter/MSTest.CoreAdapter/Execution/TypeCache.cs:649]: Have we asked @pvlakshm about precedence order?

Comment (jayaranigarg) [reply src/Adapter/MSTest.CoreAdapter/Execution/TypeCache.cs:649]: What happens when we passing runsettings arguments through commandline??

Comment (singhsarab) [reply src/Adapter/MSTest.CoreAdapter/Execution/TypeCache.cs:649]: https://github.com/Microsoft/testfx-docs/pull/40

File: test/UnitTests/MSTest.CoreAdapter.Unit.Tests/Execution/TypeCacheTests.cs (hunk 1 above)
Comment (singhsarab) [reply test/UnitTests/MSTest.CoreAdapter.Unit.Tests/Execution/TypeCacheTests.cs:825]: Done.
